### PR TITLE
Test for whitespace-only text node

### PR DESCRIPTION
**Pull request by AirQuick**
This pull request derives from #351. So needs to be handled after that.

---

This pull request just adds some tests for the feature of handling whitespace-only text nodes.
* [`xspec-space_query.xspec`](https://github.com/xspec/xspec/commit/339c8e7635a34dfac2a5b69d3f4a0f16adfa5779#diff-4e5f5c59510bbd6183dab32dfcf7d556) for XQuery
* [`xspec-space_stylesheet.xspec`](https://github.com/xspec/xspec/commit/339c8e7635a34dfac2a5b69d3f4a0f16adfa5779#diff-e9d32ce6e8edfec926e59dddad7c5032) for XSLT

### Further work
The current wiki does not mention how whitespace-only text nodes are handled. We might want to add a brief description to the wiki based on this pull request.

### CODE REVIEW

**Comment by AirQuick** [comment]
> ### Further work
> The current wiki does not mention how whitespace-only text nodes are handled. We might want to add a brief description to the wiki based on this pull request.

Done.
Added [Whitespace-only Text Nodes](https://github.com/xspec/xspec/wiki/Writing-Scenarios#whitespace-only-text-nodes) section in Wiki. I deliberately ignored `x:space` because we don't have any test for it yet.

**Comment by AirQuick** [comment]
@galtm 
This is another place `x:variable` can come in.
This pull request ensures that whitespace-only text nodes are handled in the way described in [Wiki](https://github.com/xspec/xspec/wiki/Writing-Scenarios#whitespace-only-text-nodes).

To be precise, another factor (`x:space`) affects whitespace-only text nodes. But `x:space` is not included in this pull request. It will be handled in #417 and #460.

File: test/xspec-space_stylesheet.xspec [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+				</x:param>
+			</x:call>
+			<x:expect label="Child whitespace-only text nodes are kept."
+				test="deep-equal(p/(node() except span), $wsot)" />
+			<x:expect label="Grandchild whitespace-only text nodes are kept."
+				test="deep-equal(p/span, $span-element-wsot)" />
+		</x:scenario>
+
+		<x:scenario>
+			<x:label>Elements specified by @preserve-space takes effect only on child
```
Comment (galtm) [inline test/xspec-space_stylesheet.xspec]: The grammar in the label is a little off. Alternate suggestions you might consider: `Elements specified by @preserve-space see effect only on...` or `For elements specified by @preserve-space, effect is only on...`.

**Comment by galtm** [comment]
> @galtm
> This is another place `x:variable` can come in.
> This pull request ensures that whitespace-only text nodes are handled in the way described in [Wiki](https://github.com/xspec/xspec/wiki/Writing-Scenarios#whitespace-only-text-nodes).
> 
> To be precise, another factor (`x:space`) which is not described in Wiki affects whitespace-only text nodes. But `x:space` is not included in this pull request. It will be handled in #417 and #460.
> 
> XQuery implementation in this area at the moment is not as complete as XSLT. That's why each test file is dedicated to XQuery or XSLT, and they are not shared between XQuery and XSLT.
> 
> I used `fn:deep-equal()` instead of XSpec's own feature of comparing the actual result and the expected result, because XSpec cannot compare whitespace-only text node correctly (#346) 🙄

Thanks for the background information and the wiki update! I'll add `x:variable` to these files and remove the "Check behavior of whitespace-only text node" scenario from my PR #475 after I catch up to the merges you did over the weekend.

File: test/xspec-space_stylesheet.xspec (hunk 1 above)
Comment (AirQuick) [reply test/xspec-space_stylesheet.xspec]: Thanks for the suggestions. I took the latter in ea6ee7734f5958c2c7b8bf18514779ccb64e2ede.

**Comment by AirQuick** [comment]
Thanks, @galtm.

@cirulls 
Merged this.

**Comment by galtm** [comment]
> add `x:variable` to these files and remove the "Check behavior of whitespace-only text node" scenario from my PR #475

I just uploaded 3dbc4ab41c5a5efd4f4c9467b7723c46c5144fde as part of PR #475.

**Pull request merged**